packages/core/utils/src/common/__tests__/define-config.spec.ts

```diff
@@ -958,4 +958,188 @@ describe("defineConfig", function () {
       }
     `)
   })
+
+  it("should include cloud-based modules when in cloud execution context", function () {
+    const originalEnv = { ...process.env }
+
+    process.env.EXECUTION_CONTEXT = "medusa-cloud"
+    process.env.REDIS_URL = "redis://localhost:6379"
+    process.env.S3_FILE_URL = "https://s3.amazonaws.com/medusa-cloud-test"
+    process.env.S3_PREFIX = "test"
+    process.env.S3_REGION = "us-east-1"
+    process.env.S3_BUCKET = "medusa-cloud-test"
+    process.env.S3_ENDPOINT = "https://s3.amazonaws.com"
+    const res = defineConfig({})
+
+    process.env = { ...originalEnv }
+
+    expect(res).toMatchInlineSnapshot(`
```

Comment:
suggestion: maybe we should use a regular equal here. Snapshots are super strict and any property order that changes, will break it.

Reply:
I just followed the pattern of the previous tests. I'm more than happy to change it everywhere if you think it's better.

Reply:
It is alright to keep it as it is.
This is more personal preference than anything else.